dev-client/src/screens/LocationScreens/components/LocationSoilIdCard.tsx

```diff
@@ -100,48 +87,81 @@ export const LocationPrediction = ({
   );
 };
 
-type SoilIdStatusDisplayTopMatchProps = {
-  status: SoilIdStatus;
-  topSoilMatch: LocationBasedSoilMatch | DataBasedSoilMatch | undefined;
-  t: TFunction;
+type LocationMatchDisplayProps = {coords: Coords};
+
+const LocationMatchContent = ({coords}: LocationMatchDisplayProps) => {
+  const soilIdData = useSoilIdData(coords);
+  const topSoilMatch = useMemo(() => getTopMatch(soilIdData), [soilIdData]);
+
+  return <MatchContent status={soilIdData.status} match={topSoilMatch} />;
 };
 
-const TopSoilMatchDisplay = ({
-  status,
-  topSoilMatch,
-  t,
-}: SoilIdStatusDisplayTopMatchProps) => {
+type SiteMatchDisplayProps = {coords: Coords; siteId: string};
+
+const SiteMatchContent = ({coords, siteId}: SiteMatchDisplayProps) => {
```

Comment:
question: does it make sense to break any of these components into separate files? (SiteMatchContent, MatchContent)

Reply:
I'm not sure if it does; they're all strictly internal to this one, and the `LocationMatchContent` vs `SiteMatchContent` is just a workaround to make sure that we have a valid site ID when querying the selected soil. We have a few other internal components of roughly this complexity; I think it'd make things more complex to further to subdivide this one right now.